compiler/optimizations/optimizeForallUnorderedOps.cpp

```diff
@@ -358,7 +358,7 @@ void checkLifetimesForForallUnorderedOps(FnSymbol* fn,
 
   // This runs even for fNoOptimizeForallUnordered
   // because even if the optimization is disabled, we want
-  // to mark the ends of foralls with chpl_comm_unordered_task_fence.
+  // to mark the ends of foralls with chpl_comm_unordered_consist_fence.
```

Comment:
There's also a chpl_gen_comm_getput_unordered_task_fence

Reply:
And a chpl_comm_atomic_unordered_task_fence. I'm not sure either needs to change as part of this, but might be worth it for consistency. 

I'd also be fine with changing this back to chpl_comm_unordered_task_fence. I'm not sure consist is all that more meaningful to me. (The "task" in the name implies that it only fences operations initiated by this task, it's not denoting a task end)

Reply:
Or actually `chpl_comm_getput_unordered_task_fence()`?  (Not `chpl_gen_comm_*`.)  I'll do the `_IMPL` thing with that one also.  I didn't even notice it.

Reply:
And there is also `chpl_comm_atomic_unordered_task_fence()`.  I'm happy to do that one also.  In the interest of consistency should I name them all the same?  `_task_fence` or `_consist_fence`?  Should I put them together in the source files or leave them apart (as they are now)?  Could this additional work wait?  If I can get this PR in today then I can maybe get my comm=ofi MCM conformance improvements in for the release, which would make Shasta support a bit simpler.

Reply:
There's `chpl_gen_comm_getput_unordered_task_fence` and `chpl_cache_comm_getput_unordered_task_fence` and I think `chpl_comm_getput_unordered_task_fence`. There's a bunch of these. 

There's the get/put ones in the runtime, the atomic ones in the runtime, and then this higher level unordered one that fences both puts/gets and atomics. 

What's the motivation for changing this name from `_task_fence` to `_consist_fence`? I'd probably just leave it as `_task_fence` so there's not so many names to change.

Reply:
Oh, maybe i'm misreading this (got confused reading the comments here vs the module code change)  -- You changed `chpl_comm_task_end` to `chpl_comm_unordered_consist_fence`. Maybe just call that `chpl_comm_unordered_task_fence`?

Reply:
No, I added `chpl_comm_unordered_consist_fence()` and changed some calls that referenced `chpl_comm_task_end()` to call that new one instead.  `chpl_comm_task_end()` still exists, and is still called from other places.  I'll change `chpl_comm_unordered_consist_fence()` to `chpl_comm_unordered_task_fence()`.